src/prometheus_ranch_collector.erl

```diff
@@ -247,9 +257,19 @@ convert_proc_value(status, waiting) -> 6;
 convert_proc_value(_, Value) -> Value.
 
 %% Update the metrics collection with a new metric.
--spec update_metrics(atom(), prometheus_model:'Metric'(), metrics()) -> metrics().
-update_metrics(Name, Metric, Coll) ->
-	maps:update_with(Name, fun (Old) -> [Metric|Old] end, [Metric], Coll).
+-spec update_metrics(labels(), [{atom(), number() | undefined | infinity}], metrics()) -> metrics().
+update_metrics(Labels, RawMetrics, Coll) ->
+	lists:foldl(
+		fun
+			({_, undefined}, Acc) ->
+				Acc;
+			({Name, Value}, Acc) ->
+				Metric=gauge_metric(Labels, Value),
```

Comment:
As this is now the only place where `gauge_metric` is used, the import should be removed. While you're at it, I think it would be good to remove the import for `create_mf` as well.

Reply:
Both done :)